src/pybind/mgr/cli_api/tests/test_cli_api.py

```diff
@@ -0,0 +1,26 @@
+import unittest
+
+from ..module import CLI, BenchmarkException
+
+
+class BenchmarkRunnerTest(unittest.TestCase):
+    def test_number_of_calls_on_start_fails(self):
+        with self.assertRaises(Exception) as ctx:
+            CLI.benchmark(0, 10, "get", "osd_map")
+        self.assertEqual(str(ctx.exception), "Number of calls must be greater than 0.")
+
+    def test_number_of_parallel_calls_on_start_fails(self):
+        with self.assertRaises(Exception) as ctx:
+            CLI.benchmark(10, 0, "get", "osd_map")
+        self.assertEqual(str(ctx.exception), "Number of threads must be greater than 0.")
+
+    def test_number_of_parallel_calls_on_start_works(self):
+        CLI.benchmark(10, 10, "get", "osd_map")
+
+    def test_function_name_fail(self):
+        with self.assertRaises(Exception) as ctx:
+            CLI.benchmark(10, 10, "g", "osd_map")
+        self.assertEqual(str(ctx.exception), "Could not find the public function you are requesting")
```

Comment:
```suggestion
    def test_function_name_fails(self):
        for iterations in [0, 1]:
            threads = 0 if iterations else 1
            with self.assertRaises(BenchmarkException) as ctx:
                CLI('CLI', 0, 0).benchmark(iterations, threads, 'fake_method', [])
            self.assertEqual(str(ctx.exception), "Number of Total calls and number of parallel calls must be greater than 0")

        result: HandleCommandResult = CLI('CLI', 0, 0).benchmark(1, 1, 'fake_method', [])
        self.assertEqual(result.stderr, 'Could not find the public function you are requesting')
```
Here I give you an example on how to test the failures.
Please fix the rest of the failing tests accordingly.

Reply:
I was going to suggest [`assertRaisesRegex`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex):
```suggestion
    def test_function_name_fail(self):
        with self.assertRaisesRegex(ValueError, "Could not find the public function you are requesting"):
            CLI.benchmark(10, 10, "g", "osd_map")
```
And then realized (thanks to @alfonsomthd's comment) that some errors are exposed as return messages while others are exceptions... So let's stick to a single behaviour then.